File: src/main/java/com/wgzhao/addax/admin/model/TbImpChkKey.java
package com.wgzhao.addax.admin.model;

import jakarta.persistence.Embeddable;
import lombok.Getter;
import lombok.Setter;

import java.io.Serializable;
import java.util.Date;
import java.util.Objects;

@Embeddable
@Setter
@Getter
public class TbImpChkKey implements Serializable
{
    private String chkContent;
    private Date updtDate;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        TbImpChkKey that = (TbImpChkKey) o;
        return Objects.equals(chkContent, that.chkContent) && Objects.equals(updtDate, that.updtDate);
    }

    @Override
    public int hashCode() {
        return Objects.hash(chkContent, updtDate);
    }
}

File: src/main/kotlin/com/wgzhao/addax/admin/model/TbImpChkKey.kt
package com.wgzhao.addax.admin.model

import jakarta.persistence.Embeddable

import java.util.*

/**
 * 采集校验主键实体类。
 * 用于标识校验内容和更新时间的复合主键。
 */
@Embeddable
data class TbImpChkKey(
    private val chkContent: String? = null,
    private val updtDate: Date? = null
)

